expression/typeinferer.go

```diff
@@ -388,7 +388,8 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 		ast.FoundRows, ast.Length, ast.Extract, ast.Locate, ast.UnixTimestamp, ast.Quarter, ast.IsIPv4, ast.ToDays,
 		ast.ToSeconds, ast.Strcmp, ast.IsNull, ast.BitLength, ast.CharLength, ast.CRC32, ast.TimestampDiff,
 		ast.Sign, ast.IsIPv6, ast.Ord, ast.Instr, ast.BitCount, ast.TimeToSec, ast.FindInSet, ast.Field,
-		ast.GetLock, ast.ReleaseLock, ast.Interval, ast.Position, ast.PeriodAdd, ast.PeriodDiff, ast.IsIPv4Mapped, ast.IsIPv4Compat, ast.UncompressedLength:
+		ast.GetLock, ast.ReleaseLock, ast.Interval, ast.Position, ast.PeriodAdd, ast.PeriodDiff, ast.IsIPv4Mapped, ast.IsIPv4Compat, ast.UncompressedLength,
+		ast.ASCII:
```

Comment:
without this line, i got panic with make dev.

> PANIC: executor_test.go:779: testSuite.TestStringBuiltin
> 
> ... Panic: cannot get STRING result from ClassInt expression (PC=0x1057B98)
> 
> /usr/local/go/src/runtime/panic.go:489
>   in gopanic
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/expression/builtin.go:234
>   in baseIntBuiltinFunc.evalString
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/expression/scalar_function.go:216
>   in ScalarFunction.EvalString
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/expression/scalar_function.go:188
>   in ScalarFunction.Eval
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/expression/constant_fold.go:38
>   in FoldConstant
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/plan/expression_rewriter.go:96
>   in planBuilder.rewriteWithPreprocess
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/plan/expression_rewriter.go:64
>   in planBuilder.rewrite
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/plan/logical_plan_builder.go:394
>   in planBuilder.buildProjection
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/plan/logical_plan_builder.go:1035
>   in planBuilder.buildSelect
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/plan/planbuilder.go:152
>   in planBuilder.build
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/plan/optimizer.go:67
>   in Optimize
> compiler.go:41
>   in Compiler.Compile
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/tidb.go:149
>   in Compile
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/session.go:607
>   in session.Execute
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:99
>   in TestKit.Exec
> /Users/loushuai/workspace/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:136
> 

But i can execute the select with tidb-server directly!
It confused!